client/state/data-layer/wpcom/sites/hosting/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import { HOSTING_RESTORE_DATABASE_PASSWORD } from 'state/action-types';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+
+const requestRestoreDatabasePassword = action =>
+	http(
+		{
+			path: `/sites/${ action.siteId }/hosting/restore-database-password`,
```

Comment:
For folks reading, we try to mirror the API paths with the file structure in `data-layer/wpcom`. 

The updates are looking much tidier with the port to the data-layer @mmtr thanks for updating!

Reply:
thanks! good catch - this should probably be `…wpcom/sites/hosting/restore-database-password.js`